bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/dto/TellstickLocalDeviceDTO.java

```diff
@@ -112,4 +112,23 @@ public String getType() {
     public void setType(String type) {
         this.type = type;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof TellstickLocalDeviceDTO other)) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
```

Comment:
This check seems rather pointless, as it duplicates the instance of check above?

Reply:
> This check seems rather pointless, as it duplicates the instance of check above?

There is a difference. As `instanceof` can also be a subtype, it might be a different class. Have not looked at the code to check if that might be the case in this binding. Probably not.

Reply:
My feeling is that one of the 2 tests is useless, either the first or the second, no ?
I don't know what is expected by the author.

Reply:
> My feeling is that one of the 2 tests is useless, either the first or the second, no ? I don't know what is expected by the author, either to have exactly same class for the two objects or not.

Reply:
I used the same pattern as here:
https://github.com/openhab/openhab-addons/blob/3b30d5ab11d8669e5028b3c78e42e141e0be5abd/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/xml/TellstickNetSensor.java#L135
I would say that this is an example of defensive programming, i.e. try to handle any case in a predictable manner.

Reply:
That example doesn't have the instanceof check, though...

Reply:
Instanceof and getclass are slightly different checks. If defensive programming is preferred, this is perfectly fine.
But I can’t imagine looking at the code that instanceof check passes but the getclass not. So either way would be fine and more of a personal preference regarding future fit / readability etc.

Reply:
> That example doesn't have the instanceof check, though...

Well, in my first commit it was a copy of that code then @lsiepel suggested the instanceof change which is made in commit 2.

Reply:
It is 'free' as you did both null check and casting. So you might consider dropping the:
```if (getClass() != obj.getClass()) {```
But  as said before, it is a slightly different check and more of a personall preference. I would drop it as it is not needed (at this moment).

Reply:
Ok, the class comparasion is removed in commit dc7d5e8f0e38b44a3bc93267fdda45446c04a34b